Utils improvement

alamo18 opened the pull request.

Commit:
Gradle Update dependencies (3.2.0-alpha03)

Signed-off-by: Morgan Lee <chewymorgy@gmail.com>

Commit:
Utils: Improve restartLauncher method

Signed-off-by: Morgan Lee <chewymorgy@gmail.com>

Commit:
Utils: Fix String format warning

Signed-off-by: Morgan Lee <chewymorgy@gmail.com>

divadsn reviewed (commented):
There is no point in parsing a String to a String tbh xD

Commit:
Utils: Simplify isBlacklistedAppInstalled() return statement

Signed-off-by: Morgan Lee <chewymorgy@gmail.com>

alamo18 commented:
i'm drunk, lemme undo dat

Commit:
Revert "Utils: Fix String format warning"

This reverts commit b89d312c74af61435142c50615e7181be0c5954d.

skittles9823 commented:
@alamo18 Lmao nice one Morgan.

Commit:
Utils: Clean unused imports

Signed-off-by: Morgan Lee <chewymorgy@gmail.com>

alamo18 commented:
Is this ok?

abhijeetbohra commented:
Nice one! Morgan rocks!

suphon-t commented on app/src/main/java/ch/deletescape/lawnchair/Utilities.java:
```diff
@@ -1113,19 +1108,11 @@ public static boolean checkOutdatedLawnfeed(Context context) {
+        Intent startActivity = new Intent(context, Launcher.class);
+        startActivity.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+        context.startActivity(startActivity);
+        ((Activity) context).finish();
```
Finishing and restarting an activity is not equal to restarting the launcher. There are some static references that will still persist when all activities are destroyed.

suphon-t replied:
Also, you can't just cast a Context to an Activity and expect it to always work. If you really need to, change the method signature or add a check before casting.

alamo18 replied:
ok will do after class

alamo18 commented:
Summary of my PR:
1. Gradle 3.1.0-alpha08 => 3.2.0-alpha03
2. restartLauncher improvement (start Launcher.class, clear top then finish. There is no need to import PackageManager and AlarmManager)
3. Cleanup

abhijeetbohra commented:
@alamo18 Great work!

Commit:
Utilities: Ensure app is fully killed
- System.exit(0) should kill the app properly
- check for context validity before casting Activity

Signed-off-by: Morgan Lee <chewymorgy@gmail.com>

Commit:
Update gradle (3.2.0-alpha04)

Signed-off-by: Morgan Lee <chewymorgy@gmail.com>
